### PR TITLE
Setup binary package repository for different platforms

### PR DESCRIPTION
**Pull request by walkowif**
Use `pkgcache::repo_resolve` to dynamically set proper binary package repositories.

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -49,6 +49,11 @@ inputs:
     description: Maximum number of iterations to resolve dependencies.
     required: false
     default: 10
+  repository-list:
+    description: |
+      Comma-separated list of pkgcache-compatible repository names from which the dependencies will be downloaded.
+    required: false
+    default: "https://pharmaverse.r-universe.dev,PPM@latest"
```
Comment (walkowif) [inline action.yaml:56]:
```suggestion
    default: "PPM@latest"
```

Comment (pawelru) [reply action.yaml:56]: I think I agree. If runiverse is required then we can specify it when running the action. The default should be just ppm

File: action.yaml [hunk 2]
```diff
@@ -74,6 +79,29 @@ runs:
...
+    run: |
+      R -s -q <<RSCRIPT
+      getOption("repos")
+      if (!require("pkgcache")) install.packages("pkgcache")
+      RSCRIPT
+      cat > ${HOME}/.Rprofile <<EOF
+      repository_list <- unlist(strsplit(
+        "${{ inputs.repository-list }}", split=","
+      ))
+      options(repos = c(vapply(repository_list, pkgcache::repo_resolve, character(1))))
```
[the remark this answers is not in the document]
Comment (pawelru) [reply action.yaml:92]: It should be c(vapply(...), default-cran())

**Comment by pawelru** [comment]
Code wise it's looking good now. I don't have any further comments. 
I would have to check on the test results before the final approve.

**Comment by walkowif** [comment]
Just in case, here are the latest pipelines where I was testing this:
* [R-hub workflow](https://github.com/walkowif/teal/actions/runs/11033274594) with `repository-list: PPM@latest`
* [R CMD check](https://github.com/walkowif/teal/actions/runs/11033627459/job/30645214942) with `repository-list: https://pharmaverse.r-universe.dev,PPM@latest`

**Review by pawelru — COMMENTED** [review]
OK I actually have ~one~ two more requests. 

This is what I'm seeing:
image

1. Can we make use of the `::group::` notation when printing the repos out
2. Can we remove "Repositories written to ..."  from the console output.

File: action.yaml [hunk 3]
```diff
@@ -83,8 +88,25 @@ runs:
...
+      cat > ${HOME}/.Rprofile <<EOF
+      repository_list <- unlist(strsplit(
+        "${{ inputs.repository-list }}", split=","
+      ))
+      options(repos = c(vapply(repository_list, pkgcache::repo_resolve, character(1)), getOption("repos")))
+      EOF
+      R -s -q <<RSCRIPT
+      cat("Resolved repositories loaded from .Rprofile:\n")
+      getOption("repos")
+      RSCRIPT
```
Comment (pawelru) [inline action.yaml:105]: following https://github.com/insightsengineering/setup-r-dependencies/pull/8#pullrequestreview-2328628600
do we need this?

**Review by pawelru — APPROVED** [review]
awesome 👍 
Many thanks!

**Pull request merged**